### PR TITLE
Add support for `sortByName`

### PR DESCRIPTION
**Pull request by shanson7**
Also removes the automatic sorting of output by series target in `plan.go`.

### CODE REVIEW

**Comment by shanson7** [comment]
Seems like an unrelated test failed. I have seen this test fail on me a couple of times, transiently.

**Comment by shanson7** [comment]
@DanCech - Can you comment on the sorting of graphite-web? I'm trying to look through the code, but I'm not sure where that's handled

**Comment by DanCech** [comment]
https://github.com/graphite-project/graphite-web/blob/master/webapp/graphite/render/datalib.py#L237

The series that come back from any data query are sorted by name here, running them through `sortBy*` alters the order from there.

**Comment by shanson7** [comment]
Ah, so they are ordered *before* functions are run on them?

**Comment by DanCech** [comment]
Yessir

**Comment by shanson7** [comment]
And now metrictank matches graphite in it's presort.

File: util/naturalsort_test.go [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+import (
+	"fmt"
+	"math/rand"
+	"sort"
+	"testing"
+)
+
+func shuffleSeries(slice []string) {
+	for i := len(slice) - 1; i > 0; i-- {
```
Comment (Dieterbe) [inline util/naturalsort_test.go]: why count down instead of the more commonly used count up?

Comment (shanson7) [reply util/naturalsort_test.go]: This is the most common version of the Fisher-Yates algorithm. The alternative is from 0 to len-2. I don't really care which one to use.

Comment (Dieterbe) [reply util/naturalsort_test.go]: ok then, sounds good

**Comment by Dieterbe** [comment]
thanks for another nice contribution @shanson7 ! :100:

**Pull request merged**